api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirUserDaoImpl.java

```diff
@@ -11,31 +11,55 @@
 
 import static org.hibernate.criterion.Restrictions.eq;
 
+import java.util.Optional;
+
+import ca.uhn.fhir.rest.param.StringAndListParam;
+import ca.uhn.fhir.rest.param.TokenAndListParam;
 import lombok.AccessLevel;
 import lombok.Setter;
-import org.hibernate.SessionFactory;
+import org.hibernate.Criteria;
 import org.openmrs.User;
+import org.openmrs.module.fhir2.FhirConstants;
 import org.openmrs.module.fhir2.api.dao.FhirUserDao;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Qualifier;
+import org.openmrs.module.fhir2.api.search.param.SearchParameterMap;
 import org.springframework.stereotype.Component;
 
 @Component
 @Setter(AccessLevel.PACKAGE)
-public class FhirUserDaoImpl implements FhirUserDao {
-	
-	@Autowired
-	@Qualifier("sessionFactory")
-	private SessionFactory sessionFactory;
+public class FhirUserDaoImpl extends BaseFhirDao<User> implements FhirUserDao {
 	
 	@Override
 	public User getUserByUuid(String uuid) {
-		return (User) sessionFactory.getCurrentSession().createCriteria(User.class).add(eq("uuid", uuid)).uniqueResult();
+		return (User) getSessionFactory().getCurrentSession().createCriteria(User.class).add(eq("uuid", uuid)).uniqueResult();
 	}
```

Comment:
Ditto: just using `get()` in place of duplicating this method makes the most sense to me.